homeassistant/components/tado/device_tracker.py

```diff
@@ -1,149 +1,164 @@
 """Support for Tado Smart device trackers."""
 from __future__ import annotations
 
-import asyncio
-from collections import namedtuple
-from datetime import timedelta
-from http import HTTPStatus
 import logging
-
-import aiohttp
-import voluptuous as vol
-
-from homeassistant.components.device_tracker import (
-    DOMAIN,
-    PLATFORM_SCHEMA as BASE_PLATFORM_SCHEMA,
-    DeviceScanner,
-)
-from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
-from homeassistant.core import HomeAssistant
-from homeassistant.helpers.aiohttp_client import async_create_clientsession
-import homeassistant.helpers.config_validation as cv
+from typing import Any
+
+from homeassistant.components.device_tracker import SourceType
+from homeassistant.components.device_tracker.config_entry import TrackerEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import STATE_HOME, STATE_NOT_HOME
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.issue_registry import IssueSeverity, async_create_issue
 from homeassistant.helpers.typing import ConfigType
-from homeassistant.util import Throttle
-
-_LOGGER = logging.getLogger(__name__)
-
-CONF_HOME_ID = "home_id"
-
-MIN_TIME_BETWEEN_SCANS = timedelta(seconds=30)
-
-PLATFORM_SCHEMA = BASE_PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_USERNAME): cv.string,
-        vol.Required(CONF_PASSWORD): cv.string,
-        vol.Optional(CONF_HOME_ID): cv.string,
-    }
-)
-
-
-def get_scanner(hass: HomeAssistant, config: ConfigType) -> TadoDeviceScanner | None:
-    """Return a Tado scanner."""
-    scanner = TadoDeviceScanner(hass, config[DOMAIN])
-    return scanner if scanner.success_init else None
-
-
-Device = namedtuple("Device", ["mac", "name"])
 
+from .const import DATA, DOMAIN, SIGNAL_TADO_MOBILE_DEVICE_UPDATE_RECEIVED
 
-class TadoDeviceScanner(DeviceScanner):
-    """Scanner for geofenced devices from Tado."""
+_LOGGER = logging.getLogger(__name__)
 
-    def __init__(self, hass, config):
-        """Initialize the scanner."""
-        self.hass = hass
-        self.last_results = []
 
-        self.username = config[CONF_USERNAME]
-        self.password = config[CONF_PASSWORD]
+async def async_get_scanner(hass: HomeAssistant, config: ConfigType) -> None:
+    """Configure the Tado device scanner."""
+    async_create_issue(
```

Comment:
Start the config flow at the import step here.

Example:
https://github.com/home-assistant/core/blob/3e641b3ef2b0717a61bfb4a1675ebfbe7c11bbda/homeassistant/components/tile/device_tracker.py#L63-L72

Probably await the config flow instead and check the result of the return value. If it's an abort result we want to create an issue saying the import was aborted. Otherwise we want to create an issue asking the user to remove the section in configuration.yaml.

Here's an example of the latter:
https://github.com/home-assistant/core/blob/3e641b3ef2b0717a61bfb4a1675ebfbe7c11bbda/homeassistant/components/fitbit/sensor.py#L620-L641

Reply:
Thanks for the guidance! I have pushed my changes to get this in place.
Test cases will be written if this path is approved and stable enough, to make the effort in writing tests worthwhile and durable.